gandalf.js

```diff
@@ -1,130 +1,130 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.